### PR TITLE
thorfinn: dropout regularization (dropout=0.1) on SOTA stack

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

The SOTA model uses zero dropout (`--model-dropout 0.0`, the default). With only 400 training samples, Transolver may be overfitting: the train/val gap at best epoch (ep9) is wider than expected for this model size. Stochastic depth / dropout in the Transformer blocks is a classic regularization technique that can act as an implicit ensemble over training.

The model already includes a `--model-dropout` flag (wired through to `TransformerBlock` via the attention and MLP paths). This has **never been tested** against the SOTA stack. At dropout=0.1 we are applying 10% stochastic masking inside each attention head's `proj_dropout` layer, adding mild noise without destabilizing training.

The key trade-off: dropout can hurt gradient flow when the signal is weak (small dataset with 400 samples), but DrivAerML has structured geometric inputs (not noisy natural language tokens) so the information signal per token is high. We predict a small but real regularization benefit: dropout=0.1 may reduce overfitting on the 34 val samples, pulling val_abupt below the current 9.484 SOTA.

**Prediction:** dropout=0.1 should improve validation metrics by ~1–3%, particularly on out-of-distribution axes (tau_y, tau_z) where overfitting to training-set geometry is most likely. If there is no improvement over 5 epochs, the model is not overfitting and we can retire this axis.

## Instructions

Apply the following **single change** from SOTA. Everything else is identical to PR #115.

```bash
cd target/
torchrun --standalone --nproc_per_node=8 train.py \
  --agent thorfinn \
  --optimizer lion --lr 1e-4 --weight-decay 5e-4 \
  --no-compile-model --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 --model-slices 128 \
  --model-dropout 0.1 \
  --ema-decay 0.999 \
  --wandb-group tay-round13-dropout-0p1
```

**The only change from SOTA is `--model-dropout 0.1` (was the default 0.0).** All other flags are locked.

If results at epoch 5 show val_abupt > 11.0 (clearly losing), kill the run and try `--model-dropout 0.05` with the same command (just change the value and relaunch). Report both runs in this PR.

## Expected behaviour

- Throughput: essentially unchanged (dropout is a cheap op). You should complete the full 9 epochs within the 270-min budget.
- Convergence: slightly slower early (dropout adds noise), but may land lower at ep8–ep9. If the val curve is still descending at ep9, note it.
- If you have time after the first run, try a quick second run at `--model-dropout 0.05` to bracket the optimal value. Use `--wandb-group tay-round13-dropout-0p1` for the first and `--wandb-group tay-round13-dropout-0p05` for the second.

## Baseline to beat

Current SOTA from PR #115 (thorfinn), W&B run `d03oghpp`:

| Metric | SOTA (PR #115) | AB-UPT |
|---|---:|---:|
| `val_primary/abupt_axis_mean_rel_l2_pct` | **9.484** | — |
| `test_primary/abupt_axis_mean_rel_l2_pct` | **10.580** | — |
| `test_primary/surface_pressure_rel_l2_pct` | **5.690** | 3.82 |
| `test_primary/wall_shear_rel_l2_pct` | **10.419** | 7.29 |
| `test_primary/volume_pressure_rel_l2_pct` | **12.740** | 6.08 |
| `test_primary/wall_shear_x_rel_l2_pct` | **8.908** | 5.35 |
| `test_primary/wall_shear_y_rel_l2_pct` | **12.491** | 3.65 |
| `test_primary/wall_shear_z_rel_l2_pct` | **13.071** | 3.63 |

**Beat `val_primary/abupt_axis_mean_rel_l2_pct < 9.484` to be a winner.**

## Results

Post your epoch-by-epoch `val/abupt_axis_mean_rel_l2_pct` trajectory and the final full test metrics table in a comment on this PR. If you ran both 0.1 and 0.05, include both.

### CODE REVIEW

**Comment by morganmcg1** [comment]
STUDENT thorfinn:

Heads-up before kicking off: edward already tested `--model-dropout 0.05` on the same SOTA stack in **PR #162** (closed 2026-05-01 11:01 UTC), and it regressed every metric:

| Metric | SOTA (PR #115) | dropout=0.05 (#162) | Delta |
|---|---:|---:|---:|
| val_abupt | 9.484 | 9.757 | +2.87% |
| test_abupt | 10.580 | 11.029 | +4.24% |
| surface_pressure | 5.690 | 5.820 | +2.29% |
| wall_shear | 10.419 | 10.842 | +4.06% |
| volume_pressure | 12.740 | 12.853 | +0.89% |

W&B run for #162: `tfumujfi`. The val/test ratio worsened from 1.115 to 1.130 — i.e. dropout did not improve generalization.

Plan:
- Proceed with `--model-dropout 0.1` as instructed (this is the higher-dropout bracket point, untested).
- I will **not** re-run 0.05 if 0.1 also loses; we already have edward's run as the lower bracket point. I'll cite #162 in the results instead.
- If 0.1 surprisingly improves val_abupt, I'll flag that immediately as a non-monotonic outcome worth deeper investigation.

Launching now.

**Comment by morganmcg1** [comment]
STUDENT thorfinn:

Note: PR #242 was merged anomalously while `thorfinn/dropout-0p1` was still mid-epoch-2 (run f8xqojs2/group tay-round13-dropout-0p1, launched 18:00 UTC, no validation epochs completed yet). I'm killing the in-flight training now to free GPUs for newly assigned PR #247 (lr-cosine-t-max=14). No final val/test metrics for dropout=0.1 — the run was still in epoch 2/50 when terminated. If you want this re-run cleanly, please reassign or open a new PR.
